### PR TITLE
Use TBB oneAPI

### PR DESCRIPTION
**Pull request by simogasp**
This PR is a rework of #178 to enable building with the new TBB oneAPI.
Contrary to the previous, this breaks the compatibility with previous versions of TBB, as the cmake now uses CONFIG.
Major changes:

* update cmake for finding and linking with TBB oneAPI
* code fixes to replace deprecated tbb mutex and locks, using those from the std library
* update the docker configuration files
* update the CI
* update the doc

When merged we can make a patch release and submit a PR to vcpgk to bump the version as well. These changes are already taken into account in vcpkg by patches that do more or less the same thing.

### CODE REVIEW

**Comment by simogasp** [comment]
@p12tic there still 1 issue that I have to commit and it is related to this
https://github.com/Warlockbugs/cmangos-authserver/commit/e09dcc98afe0dc3413c668297b8f79d7bb2989bf
Do you have any advice on the best way to do it? Is it better to add a
```c++
#ifndef _GNU_SOURCE
#define _GNU_SOURCE
#endif
```
or is it better to add the definition in cmake with `target_compile_definitions`? Both work, I just don't know if there is a preferable way.

**Comment by p12tic** [comment]
@simogasp I didn't see any problems with boost + aliceVision on macOS. What exact symptoms are you seeing?

**Comment by simogasp** [comment]
I don't know if it depends on the version, but (on Silicon) I had a similar compiling error as https://github.com/boostorg/stacktrace/issues/88
using boost 1.76

**Comment by simogasp** [comment]
and they add this patch on vcpkg to the cctag port as well when they bumped the tbb version

**Comment by p12tic** [comment]
Interesting, I'm also on Apple arm64 and boost 1.76. It could be that the reason why I'm not seeing problems is that currently I'm compiling macports environment which likely has different libraries.

**Comment by p12tic** [comment]
Ah, I'm not actually compiling cctag component on macOS, sorry for the noise (facepalm).

Anyway, defining _GNU_SOURCE in this case shouldn't cause any problems.

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
@simogasp Appveyor build fails because of incorrect expected archive hash.

**Comment by simogasp** [comment]
> @simogasp Appveyor build fails because of incorrect expected archive hash.

yes it must be the cache, we should move to github actions like AV where we can keep control of vcpkg version, even though is less annoying here given the relatevelt fewer dependencies

**Comment by simogasp** [comment]
ah yes this breaks the appveyor CI as oneAPI is not yet in the version of vcpkg shipped with the appveyor image. 
We need to update vcpkg to a more recent version containing the oneAPI like it is explained here
https://www.appveyor.com/docs/lang/cpp/#vc-packaging-tool

File: src/cctag/utils/Exceptions.hpp [hunk 1]
```diff
@@ -5,9 +5,11 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-#ifndef _CCTAG_EXCEPTIONS_HPP_
-#define _CCTAG_EXCEPTIONS_HPP_
+#pragma once
 
+#ifndef _GNU_SOURCE
```
Comment (fabiencastan) [inline src/cctag/utils/Exceptions.hpp:13]: Need a comment with links to explain why we have that here.

**Review by fabiencastan — CHANGES_REQUESTED** [review]
Great PR. Just asking for one small change.

File: src/cctag/utils/Exceptions.hpp (hunk 1 above)
Comment (simogasp) [reply src/cctag/utils/Exceptions.hpp:13]: done

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**